connectors/es/index.py

```diff
@@ -40,6 +40,18 @@ def _create_object(self, doc):
         """
         raise NotImplementedError
 
+    async def fetch_by_id(self, id):
+        await self.client.indices.refresh(index=self.index_name)
+
+        try:
+            resp = await self.client.get(index=self.index_name, id=id)
+        except ApiError as e:
+            logger.critical(f"The server returned {e.status_code}")
+            logger.critical(e.body, exc_info=True)
```

Comment:
this does not raise any error, and returns None

we need to raise since the user of this method make the expectation a connector is returned